src/devSVG.cpp

```diff
@@ -82,39 +91,63 @@ inline void write_escaped(FILE* f, const char* text) {
   }
 }
 
-inline void write_attr_col(FILE* f, const char* attr, int col) {
+inline void write_attr_dbl(FILE* f, const char* attr, double value) {
+  fprintf(f, " %s='%.2f'", attr, value);
+}
+
+inline void write_attr_str(FILE* f, const char* attr, const char* value) {
+  fprintf(f, " %s='%s'", attr, value);
+}
+
+
+
+// Beginning of writing style attributes
+inline void write_style_begin(FILE* f) {
+  fputs(" style='", f);
+}
+
+// End of writing style attributes
+inline void write_style_end(FILE* f) {
+  fputs("'", f);
+}
+
+// Writing style attributes related to colors
+inline void write_style_col(FILE* f, const char* attr, int col, bool head_space = true) {
```

Comment:
How about instead of `head_space`, call it `first`? (because you don't need a space before the first element)